tests/sentry/api/endpoints/test_project_users.py

```diff
@@ -108,3 +108,28 @@ def test_ip_search(self):
 
         assert response.status_code == 200, response.content
         assert len(response.data) == 2
+
+    def test_delete_event_user(self):
+        # Only delete an event user as a superuser
+        self.login_as(user=self.user, superuser="true")
+
+        assert EventUser.objects.count() == 2
+
+        response = self.client.delete(f"{self.path}?query=id:{self.euser2.ident}")
+
+        assert response.status_code == 200
+        assert EventUser.objects.count() == 1
+
+        # Only allow deletion if you query by `id`
+        response = self.client.delete(f"{self.path}?query=email:foo@example.com")
+
+        assert response.status_code == 500
```

Comment:
Well formed requests from users shouldn't ever 500. Instead they should return with 4xx response code letting the user know what mistake they made.